app/assets/javascripts/offendersummary/components/nextAppointment.js

```diff
@@ -0,0 +1,110 @@
+import React, {Component, Fragment} from 'react';
+import * as PropTypes from 'prop-types';
+import ErrorMessage from './errorMessage'
+import moment from 'moment'
+import {staff} from '../../helpers/offenderManagerHelper'
+import {dateFromISO} from '../../helpers/formatters'
+
+class NextAppointment extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    componentWillMount() {
+        const { getNextAppointment } = this.props;
+        getNextAppointment();
+    }
+
+    componentDidUpdate() {
+        window.GOVUKFrontend.initAll();
```

Comment:
Just a thought but we could replace this with a node ref callback and then call `new Details(detailRef).init()` - it might improve readability and negate any side affects `initAll` might have. TBH I'm not sure what is best.

Reply:
I don't like using _componentDidUpdate_ for this, especially for nested components but it's a tricky thing as we do need to trigger the GOV.UK JavaScript.

I totally agree that in components like this we should be initialising only the used GOV.UK components - maybe the best way is to remove the "catch all" _componentDidUpdate_ within _offenderSummaryPage.js_ and make each component handle the init method requirements.

Reply:
Yes, give me 30 minutes or and I see if I can do this a bit better..